docs/40_smart-contract-guides/80_linking-custom-permission.md

```diff
@@ -2,72 +2,479 @@
+This tutorial requires the following:
 
-With great power comes great responsibility. This functionality poses some challenges to the security of your contract and its users. Ensure you understand the concepts and steps prior to putting them to use.
+* Access to a running blockchain. Click on this link for instructions on [running a blockchain](01_before-you-begin/10_running-a-blockchain.md)
+* An EOSIO account and access to the account keys. Click on this link for information on [Accounts and Permissions](01_before-you-begin/20_accounts-and-permissions.md)
+* Some knowledge of smart contracts, we recommend following the [hello world tutorial](../30_getting-started-guide/25_hello-world.md) tutorial
 
-[[info |Parent permission ]]
-| When you create a custom permission, the permission will always be created under a parent permission.
+## What is a Custom Permission
 
-If you have the authority of a parent permission which a custom permission was created under, you can always execute an action which requires that custom permission.
+When an account is created two permissions are created by default; `owner` and `active`. You can create a new permission, a custom permission, as a child permission of `owner`, `active` or another custom permission. Custom permissions require a public and private key pair. These custom permissions may be linked to an action of a smart contract and specify the permission required to execute that action. This permission system enables accounts to have flexible and highly granular control over accounts and smart contract actions.
 
-## Step 1. Create a Custom Permission
-Firstly, let's create a new permission level on the `alice` account:
+[[info]]
+| The permission `eosio.code` is a special permission which allows smart contracts to call inline actions. Smart contracts do not have access to account keys and the EOSIO blockchain performs an authorization check when an inline action is called. Adding `eosio.code` to an account permission gives explicit permission to execute code. For information on how to add the `eosio.code` see [cleos set account permission](https://developers.eos.io/manuals/eos/latest/cleos/command-reference/set/set-account-permission.md)
+
+## Why use a Custom Permission
+
+Use custom permssions to improve the security on your account by setting a custom permission for calling specific transactions. This will limit the need to use the `owner` and `active` keys and limit the risk of these permissions being compromised.
+
+[[info]]
+| Custom permissions always have a parent permission. The parent permission has the authority to execute the same transactions as the custom permission, and to recover the custom permission keys if that custom permission is compromised.
+
+## Using Custom Permissions
+
+The following examples show you how to:
+
+* Create a Custom Permission
+* Link a Custom Permission
+* Unlink a Custom Permission
+* Delete a Custom Permission
```

Comment:
I think the deleting a custom permission is missing, the tutorial stops after unlinking a custom permission

Reply:
nice to have suggestion: each of these 4 sub-items listed could be links to subsections of the tutorial

Reply:
added links and removed delete